theano/tensor/basic.py

```diff
@@ -5943,13 +5954,45 @@ def infer_shape(self, node, shapes):
 
 
 def diagonal(a, offset=0, axis1=0, axis2=1):
+    """
+    A helper function for Diagonal Op.
```

Comment:
I would be more explicit here. Look at how detailed [numpy's documentation](http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.diagonal.html) is.